src/components/PopoverMenu.tsx

```diff
@@ -130,17 +181,29 @@ function PopoverMenu({
     const onModalHide = () => {
         setFocusedIndex(-1);
         if (selectedItemIndex.current !== null) {
-            menuItems[selectedItemIndex.current].onSelected();
+            currentMenuItems[selectedItemIndex.current].onSelected();
             selectedItemIndex.current = null;
         }
     };
 
+    useEffect(() => {
+        if (menuItems.length === 0) {
+            return;
+        }
+        setEnteredSubMenuIndexes([]);
+        setCurrentMenuItems(menuItems);
+    }, [menuItems]);
```

Comment:
After updating the `menuItems` we should have updated the `focusedIndex` as well, otherwise we'd have two highlighted items: the new `menuitems` selected item and the stale `focusedIndex` (from the previous selected item).

Coming from https://github.com/Expensify/App/issues/50224